### ISSUE TITLE
Support non-blocking reads.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [byzhang/terrastore#1]
```
Provide non-coherent reads support, to allow reading (possibly stale)
values without acquiring a read lock.
```

Original issue reported on code.google.com by `sergio.b...@gmail.com` on 27 Nov 2009 at 4:34

**Comment by GoogleCodeExporter** [comment]
Original comment by `sergio.b...@gmail.com` on 1 Dec 2009 at 10:01
- Added labels: **Release-0.2**

**Comment by GoogleCodeExporter** [comment]
Original comment by `sergio.b...@gmail.com` on 1 Dec 2009 at 10:02
- Added labels: **Milestone-0.2**

**Comment by GoogleCodeExporter** [comment]
Original comment by `sergio.b...@gmail.com` on 1 Dec 2009 at 10:02
- Removed labels: **Release-0.2**

**Comment by GoogleCodeExporter** [comment]
Original comment by `sergio.b...@gmail.com` on 10 Dec 2009 at 9:12
- Added labels: **Priority-Medium**, **Milestone-0.3**
- Removed labels: **Priority-High**, **Milestone-0.2**

**Comment by GoogleCodeExporter** [comment]
Original comment by `sergio.b...@gmail.com` on 23 Dec 2009 at 4:22
- Added labels: **Milestone-0.4**
- Removed labels: **Milestone-0.3**

**Comment by GoogleCodeExporter** [comment]
```
Non-blocking (lock-free) reads can be performed without sacrificing consistency
thanks to the fixed partitioning scheme.
Read operations will be so lock-free *and* coherent.
```

Original comment by `sergio.b...@gmail.com` on 6 Jan 2010 at 3:44
- Changed title: **Support non-blocking reads.**
- Added labels: **Milestone-0.3.1**
- Removed labels: **Milestone-0.4**

**Comment by GoogleCodeExporter** [comment]
```
Implemented in 0.3.1.
```

Original comment by `sergio.b...@gmail.com` on 7 Jan 2010 at 10:10
- Changed state: **Fixed**
